src/spec.ts

```diff
@@ -79,10 +79,12 @@ export interface GenericUnitSpec<M, E extends Encoding> extends BaseSpec {
 
 export type UnitSpec = GenericUnitSpec<Mark, Encoding>;
 
+export type LayeredUnitSpec = GenericUnitSpec<AnyMark, Encoding>;
+
 
 export type FacetedUnitSpec = GenericUnitSpec<AnyMark, EncodingWithFacet>;
 
-export interface GenericLayerSpec<M, E> extends BaseSpec {
+export interface GenericLayerSpec<U> extends BaseSpec {
```

Comment:
You can put constraints on generics. See https://www.typescriptlang.org/docs/handbook/generics.html

`GenericLayerSpec<U extends BaseSpec>`?